sdk/main/src/TransactionReceiptQuery.cc

```diff
@@ -64,10 +65,57 @@ TransactionReceipt TransactionReceiptQuery::mapResponse(const proto::Response& r
 }
 
 //-----
-std::function<grpc::Status(grpc::ClientContext*, const proto::Query&, proto::Response*)>
-TransactionReceiptQuery::getGrpcMethod(const std::shared_ptr<internal::Node>& node) const
+Status TransactionReceiptQuery::mapResponseStatus(const proto::Response& response) const
 {
-  return node->getGrpcQueryMethod(proto::Query::QueryCase::kTransactionGetReceipt);
+  return STATUS_MAP.at(response.transactiongetreceipt().header().nodetransactionprecheckcode());
+}
+
+//-----
+typename Executable<TransactionReceiptQuery, proto::Query, proto::Response, TransactionReceipt>::ExecutionStatus
+TransactionReceiptQuery::shouldRetry(Status status, const Client& client, const proto::Response& response)
+{
+  if (const Executable<TransactionReceiptQuery, proto::Query, proto::Response, TransactionReceipt>::ExecutionStatus
```

Comment:
what exactly is `baseStatus`? I'm not clear on what logic is happening here

Reply:
This is calling the `shouldRetry` (now `determineStatus`) of the base `Executable` class before doing this specific derived class behavior. Basically it's setup this way because the `Executable::determineStatus` contains generic cases that all `Transaction`s and `Query`s should account for and check first and if those are good (i.e. `Executable::determineStatus` returns `ExecutionStatus::UNKNOWN`), and then check its individual cases.